### PR TITLE
Some enhancements and all OS support

### PR DESCRIPTION
**Pull request by ryands17**
This PR is for the following
1) Adds a noscript tag and all the styles in it.
2) Adds the preload to all style tags regardless of OS.
3) Add the ability to provide a custom bundle file name (in case of other SPA frameworks like Vue, the bundle name is often different). This option can be added using the `bundleName` property.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "html-minifier": "^3.5.5",
     "minimalcss": "^0.3.1",
     "mkdirp": "^0.5.1",
-    "puppeteer": "^0.11.0",
+    "puppeteer": "^0.13.0",
```
Comment (stereobooster) [inline package.json]: this is blocked by https://github.com/peterbe/minimalcss/pull/42

File: index.js [hunk 2]
```diff
@@ -32,7 +32,8 @@ const defaultOptions = {
   waitFor: false,
   externalServer: false,
   // workaround for https://github.com/geelen/react-snapshot/issues/66#issuecomment-331718560
-  fixWebpackChunksIssue: false, // experimental
+  fixWebpackChunksIssue: true, // experimental
+  bundleName: "main",
```
Comment (stereobooster) [inline index.js]: there are some changes in this code in https://github.com/stereobooster/react-snap/pull/41. There will be conflicts

File: index.js [hunk 3]
```diff
@@ -239,12 +248,15 @@ const fixWebpackChunksIssue = ({ page, basePath, asyncJs }) => {
...
     basePath,
+    bundleName,
     asyncJs
   );
 };
 
 const saveAsHtml = async ({ page, filePath, options, route }) => {
-  const content = await page.content();
+  let content = await page.content();
+  content = content.replace(/data\-onload/g, 'onload');
```
Comment (stereobooster) [inline index.js]: common name. Maybe use something more specific, like `react-snap-onload`

File: index.js (hunk 2 above)
Comment (ryands17) [reply index.js]: so which is the latest branch that I should pull?

File: index.js (hunk 3 above)
Comment (ryands17) [reply index.js]: Yes I can add this

File: package.json (hunk 1 above)
Comment (ryands17) [reply package.json]: So should it remain 0.11.0 then?

Comment (stereobooster) [reply package.json]: yes for now

File: index.js (hunk 2 above)
Comment (stereobooster) [reply index.js]: I suppose you can do PR against `breaking-changes` branch. It is pretty huge right now I do not want to introduce merge conflicts

Comment (ryands17) [reply index.js]: So should I merge the breaking-changes branch into my branch?

Comment (stereobooster) [reply index.js]: can you reintroduce changes to new branch based on `breaking-changes`. Merge will be messy.

Comment (ryands17) [reply index.js]: Why did you remove the 404 route from include?

Comment (stereobooster) [reply index.js]: it is added automatically if there is more than one page

**Comment by stereobooster** [comment]
Ported fix for ciritcal css https://github.com/stereobooster/react-snap/pull/48

**Pull request closed without merging**